gateway/hail.nginx.conf.in

```diff
@@ -35,6 +35,9 @@ server {
 server {
     server_name scorecard.@domain@;
 
+    gzip on;
+    gzip_types  text/plain text/html application/json;
```

Comment:
gzip is on by default, see `/etc/nginx/nginx.conf`, but the gzip_types is empty.  I think you should edit that file to enable the types we want globally.

Reply:
Got it. Do we want a single global types definition, or application-controlled definitions? For instance notebook-api never serves html.

Reply:
Global, I would assume.  I don't see a problem with having types in gzip_types that aren't served.  The question is does one service want to compress a type that another one does not.  I don't see why that would be the case.

Reply:
To not incur the cost of compression/decompression when you know your responses are small, and to avoid having to modify a global config every time you want to add a new type (for instance a font, or .ico; I can modify the server block for the nginx instance that sits in front of my service, rather than modifying the nginx server that sits at the public/private boundary in our modular nginx world). For small items, gzipping increases size (i.e 0 bytes -> ~600 bytes)

Reply:
Then set gzip_min_length.

Reply:
> to avoid having to modify a global config

I don't see what's wrong with that.  In fact, I see it as preferable to editing every server block that might use that modification.  And, as you know, I'd like make gateway service-generic so there won't be a per-service block.

Reply:
I don't have a strong opinion on this. A global option will be set in next commit, and overridden in server blocks if needed.

Reply:
Why would it need to be overridden?